arangod/Replication2/ReplicatedLog/Algorithms.cpp

```diff
@@ -489,63 +489,87 @@ auto ParticipantStateTuple::isFailed() const noexcept -> bool {
   return failed;
 };
 
+auto ParticipantStateTuple::lastTerm() const noexcept -> LogTerm {
+  return lastAckedEntry.term;
+}
+
+auto ParticipantStateTuple::lastIndex() const noexcept -> LogIndex {
+  return lastAckedEntry.index;
+}
+
 auto operator<=>(ParticipantStateTuple const& left,
                  ParticipantStateTuple const& right) noexcept {
   // return std::tie(left.index, left.id) <=> std::tie(right.index, right.id);
   // -- not supported by apple clang
-  if (auto c = left.index <=> right.index; c != 0) {
+  if (auto c = left.lastAckedEntry <=> right.lastAckedEntry; c != nullptr) {
```

Comment:
I think we still want to sort by index only. Not sure if this is still correct when we first sort by term and then by index. 🤔